examples/device/net_lwip_webserver/src/usb_descriptors.c

```diff
@@ -155,15 +155,16 @@ static uint16_t _desc_str[32];
 // Application return pointer to descriptor, whose contents must exist long enough for transfer to complete
 uint16_t const* tud_descriptor_string_cb(uint8_t index, uint16_t langid)
 {
-  (void)langid;
+  (void) langid;
 
-  unsigned chr_count = 0;
+  uint8_t chr_count = 0;
 
   if (STR_LANGID == index)
   {
     memcpy(&_desc_str[1], string_desc_arr[0], 2);
     chr_count = 1;
   }
+#if CFG_TUD_NET == OPT_NET_ECM
   else if (STR_MAC == index)
```

Comment:
Maybe also add an ifdef for STR_MAC on line 87 in the enum definition?

Reply:
yes, it is a better. Though, thanks to this we are able to catch the `&` issue :)

Reply:
This has uncovered a coding goof on my part.  My intent was there, but I never finished improving the code.  I do NOT like 'magic numbers' in code, and my intent was for the enum to replace these 'magic numbers'; it makes things self-documenting.  The enum definition should have gone towards the top of the code, and then it would get used in "desc_device".  .iSerialNumber would get assigned STR_MAC (not the present erroneous 3 == STR_ITFNAME).  With that corrected, there is no point in the #if in tud_descriptor_string_cb(), as it is always used.  Also, "string_desc_arr[0]" should become "string_desc_arr[STR_LANGID]".

I'm not a git guru; is there a way for me to submit these changes to this PR, or will the above verbiage have to suffice?  Thanks!

Reply:
@majbthrd no hurry, once this is merged, you could just do another PR. It is much easier that way. I am not good with git as well :) . Currently in the middle of working on another branch, will switch back and finish this asap.

Reply:
No great way for you to write to this PR.... It'd require that @hathach give you write permissions to his PR branch. You could, however, branch off of this PR branch, and link to that or create a PR from that.

Reply:
Seem like I have no ways to play lazy anymore.